### DISCUSSION TITLE
Convert String to MDX / React component

### DISCUSSION THREAD

**Discussion by MarquesCoding** [mdx-js/mdx#2356] [Q&A] [answered]
Hi,

So I've got some MDX (string) stored within my DB that is returned on an endpoint. What would be the best approach (because I can't figure it out) on displaying this? Is there a component that can convert the string to a markdown-ish react component? I'm currently running React & vite as my stack 🤔.

Thanks 🙏

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @MarquesCoding! 👋

> What would be the best approach?

The best approach depends on your needs.
MDX supports both server side and client side rendering.
There are guides for client side and server side rendering for a variety of frameworks in https://mdxjs.com/docs/getting-started/

> I'm currently running React & vite as my stack 🤔.

* client side rendering in React: https://mdxjs.com/packages/mdx/#evaluatefile-options
* vite guide: https://mdxjs.com/docs/getting-started/#vite
